File: Solutions/2971. Find Polygon With the Largest Perimeter/2971.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        total = 0
        ans = -1
        print(nums)
        for i in nums:
            if total > i:
                print(i)
                ans = total + i
            total += i
        return ans

File: Solutions/2971. Find Polygon With the Largest Perimeter/2971.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        long total = 0;
        long ans = -1;
        for(int i: nums){
            if (total > i){
                ans = total + i;
            }
            total += i;
        }
        return ans;
    }
}